### PR TITLE
Add Paramount+ Support 

### PR DESCRIPTION
**Pull request by IF64**
Adds Support for the Paramount+ streaming service

image

image

image

image

image

image

image

image

image

image

image

image

image

image

### CODE REVIEW

**Comment by IF64** [comment]
Nice try, I'm still going to bed.

File: websites/P/Paramount+/presence.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
+const presence = new Presence({
+  clientId: "821433038335377418"
+});
+(async function () {
```
Comment (ririxidev) [inline websites/P/Paramount+/presence.ts:4]: delete this async function, why do you need it tho? If you need to await strings, do it later in presence.on()

File: websites/P/Paramount+/presence.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  presence.on("UpdateData", () => {
+
+    vidArea = document.querySelector(".video__player-area");
+    let video: HTMLVideoElement = null,
+      details = undefined,
+      state = undefined,
+      smallImageKey = undefined,
+      smallImageText = undefined,
+      startTimestamp = undefined,
+      endTimestamp = undefined;
```
Comment (ririxidev) [inline websites/P/Paramount+/presence.ts:29]: just do presenceData object here and just assign things or remove them by ``delete``

File: websites/P/Paramount+/presence.ts [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+    const data: PresenceData = {
+      details: details,
+      state: state,
+      largeImageKey: "logo",
+      smallImageKey: smallImageKey,
+      smallImageText: smallImageText,
+      startTimestamp: startTimestamp,
+      endTimestamp: endTimestamp
+    };
```
Comment (ririxidev) [inline websites/P/Paramount+/presence.ts:231]:
```javascript
const data: PresenceData = {
      details,
      state,
      largeImageKey: "logo",
      smallImageKey,
      smallImageText,
      startTimestamp,
      endTimestamp
    };
``` better way if you have same variables named and want to assign them.

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @IF64!
You should check out thing(s) that I requested to change in my review. Please look at them!
 - Requested changes.
 - Revert yarn.lock changes.

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

**Comment by Bas950** [comment]
Fix your fork, you are editing a fack ton of presences

**Pull request closed without merging**